src/flag_gems/utils/pointwise_dynamic.py

```diff
@@ -132,627 +167,536 @@ def num_output_tensors(self) -> int:
-    code.writeline("from flag_gems.utils.type_utils import type_promotion")
-    code.writeline("import torch._prims_common as utils")
-    code.newline()
-    code.newline()
-    return code
-
-
-def generate_functional_pointwise_wrapper(
-    op_desc: OPDesc,
-    wrapper_name: str,
-    destination_passing_func_name: str,
-    code: IndentedBuffer,
-) -> IndentedBuffer:
-    # wrapper signature
-    parameters: str = parameter_for_wrapper(op_desc, include_outputs=False)
-    wrapper_signature: str = f"def {wrapper_name}({parameters}):"
-    code.writeline(wrapper_signature)
-
-    with code.indent():
-        # docstring
-        wrapper_docstring = docstring_for_functional_wrapper(op_desc)
-        code.writeline(wrapper_docstring)
-
-        shapes_str = ", ".join(
-            f"in{i}.shape" for i in range(op_desc.num_input_tensors())
-        )
-        code.writeline(f"shape = broadcast_shapes([{shapes_str}])")
+            # signature: output ptrs
+            for i in range(schema.num_outputs()):
+                code.writeline(
+                    f"out{output_tensor_index}_ptr: tl.tensor, # of tl.pointer_type"
+                )
+                output_tensor_index += 1
+
+            # signature: strides, for each tensor arguments
+            ndim = self.ndim
+            if ndim > 0:
+                # strides for inputs
+                for i in range(schema.num_input_tensors()):
+                    stride_args = _cs(f"in{i}_stride{j}: int" for j in range(ndim))
+                    code.writeline(f"{stride_args}, # strides for in{i}")
+                    stride_order_args = _cs(
+                        f"in{i}_stride_order{j}: tl.constexpr" for j in range(ndim)
+                    )
+                    code.writeline(f"{stride_order_args}, # stride order for in{i}")
+
+                # strides for outputs
+                for i in range(schema.num_output_tensors()):
+                    stride_args = _cs(f"out{i}_stride{j}: int" for j in range(ndim))
+                    code.writeline(f"{stride_args}, # strides for out{i}")
+                    stride_order_args = _cs(
+                        f"out{i}_stride_order{j}: tl.constexpr" for j in range(ndim)
+                    )
+                    code.writeline(f"{stride_order_args}, # stride order for out{i}")
+
+                # task space, used to reconstruct multi index
+                task_space_args = _cs(f"s{i}: int" for i in range(ndim))
+                code.writeline(f"{task_space_args}, # task_space")
+
+                # number of tasks, used to compute mask
+                code.writeline("num_tasks: int,")
+
+            # tile size & tiles_per_cta, gsl style
+            if ndim > 0:
+                code.writeline("tiles_per_cta: int,")
+                tile_sizes = _cs(f"tile_size{i}: tl.constexpr" for i in range(ndim))
+                code.writeline(f"{tile_sizes},")
+                code.writeline("one_tile_per_cta: tl.constexpr,")
```

Comment:
why here need `one_tile_per_cta`? In which scenarios does the presence of the tile loops impact performance?

Reply:
I will test it.

Reply:
The performance differences caused by branching are roughly similar; theoretically, the more complex the pointwise operation, the smaller this impact becomes. Is it necessary to remove the `one_tile_per_cta` option?

Reply:
We can probably remove it but retaining it now does not hurt performance. We may remove it after more tests.